src/Judopay/DataType.php

```diff
@@ -75,6 +77,29 @@ public static function coerce($targetDataType, $value)
             case static::TYPE_PRIMARY_ACCOUNT_DETAILS:
                 $primaryAccountDetails = PrimaryAccountDetails::factory($value);
                 return $primaryAccountDetails->toObject();
+
+            case static::TYPE_THREE_D_SECURE_TWO:
+                // Provided value for mandatory authenticationSource part of the available lists?
+                if (strcasecmp($value['authenticationSource'], "Unknown") != 0
```

Comment:
I don't think we want to allow a value of Unknown to be sent it (PartnerApi will create a value of Unknown if anything unexpected is received at the API level), but it's not an expected value